x-pack/plugins/monitoring/server/lib/logstash/get_nodes.ts

```diff
@@ -81,20 +81,28 @@ export async function getNodes(req: LegacyRequest, { clusterUuid }: { clusterUui
   const start = moment.utc(req.payload.timeRange.min).valueOf();
   const end = moment.utc(req.payload.timeRange.max).valueOf();
 
+  const query = createQuery({
+    type,
+    dsDataset: `${moduleType}.${dataset}`,
+    metricset: dataset,
+    start,
+    end,
+    clusterUuid,
+    metric: LogstashMetric.getMetricFields(),
+  });
+
+  query.bool.filter.push({
+    exists: {
+      field: 'logstash_stats.logstash.uuid',
+    },
+  });
```

Comment:
nit: `createQuery` [allows additional filters to be passed](https://github.com/elastic/kibana/blob/main/x-pack/plugins/monitoring/server/lib/create_query.ts#L62) as argument

Reply:
Oh nice! I'd missed that among all the other options. Thanks!